subcell_pipeline/analysis/dimensionality_reduction/fiber_data.py

```diff
@@ -0,0 +1,242 @@
+import pandas as pd
+from io_collection.keys.check_key import check_key
+from io_collection.load.load_dataframe import load_dataframe
+from io_collection.save.save_dataframe import save_dataframe
+from io_collection.save.save_json import save_json
+
+
+def get_merged_data(
+    bucket: str,
+    series_name: str,
+    condition_keys: list[str],
+    random_seeds: list[int],
+    align: bool = True,
+) -> pd.DataFrame:
+    """
+    Load or create merged data for given conditions and random seeds.
+
+    If merged data (aligned or unaligned) already exists, load the data.
+    Otherwise, iterate through the conditions and seeds to merge the data.
+
+    Parameters
+    ----------
+    bucket
+        Name of S3 bucket for input and output files.
+    series_name
+        Name of simulation series.
+    condition_keys
+        List of condition keys.
+    random_seeds
+        Random seeds for simulations.
+    align
+        True if data should be aligned, False otherwise.
+
+    Returns
+    -------
+    :
+        Merged data.
+    """
+
+    align_key = "all_samples_aligned" if align else "all_samples_unaligned"
+    data_key = f"{series_name}/analysis/{series_name}_{align_key}.csv"
+
+    # Return data, if merged data already exists.
+    if check_key(bucket, data_key):
+        print(
+            f"Dataframe [ { data_key } ] already exists. Loading existing merged data."
+        )
+        return load_dataframe(bucket, data_key, dtype={"key": "str"})
+
+    all_samples: list[pd.DataFrame] = []
+
+    for condition_key in condition_keys:
+        series_key = f"{series_name}_{condition_key}" if condition_key else series_name
+
+        for seed in random_seeds:
+            print(f"Loading samples for [ {condition_key} ] seed [ {seed} ]")
+
+            sample_key = f"{series_name}/samples/{series_key}_{seed:06d}.csv"
+            samples = load_dataframe(bucket, sample_key)
+            samples["seed"] = seed
+            samples["key"] = condition_key
+
+            if align:
+                align_fibers(samples)
+
+            all_samples.append(samples)
+
+    samples_dataframe = pd.concat(all_samples)
+    save_dataframe(bucket, data_key, samples_dataframe, index=False)
+
+    return samples_dataframe
+
+
+def align_fibers(data: pd.DataFrame) -> None:
+    """
+    Align fibers for each time point in the data.
+
+    Parameters
+    ----------
+    data
+        Simulated fiber data.
+    """
+
+    aligned_fibers = []
+
+    for time, group in data.groupby("time", sort=False):
+        coords = group[["xpos", "ypos", "zpos"]].values
+
+        if time == 0:
+            fiber = coords
+        else:
+            fiber = align_fiber(coords)
+
+        aligned_fibers.append(fiber)
+
+    all_aligned_fibers = np.vstack(aligned_fibers)
+
+    data["xpos"] = all_aligned_fibers[:, 0]
+    data["ypos"] = all_aligned_fibers[:, 1]
+    data["zpos"] = all_aligned_fibers[:, 2]
+
+
+def align_fiber(coords: np.ndarray) -> np.ndarray:
```

Comment:
This function aligns the point furthest from the x-axis along the y-axis, i.e. if the furthest point is `(0,0,1)`, it gets transformed to `(0, 1, 0)` with an angle of `pi / 2`.

It might be worth mentioning the y-axis alignment (or the fact that this minimizes the z-coordinates) in the documentation somewhere.